os/android/gralloc1bufferhandler.cpp

```diff
@@ -129,15 +133,43 @@ void Gralloc1BufferHandler::CopyHandle(HWCNativeHandle source,
   CopyBufferHandle(source, target);
 }
 
-void *Gralloc1BufferHandler::Map(HWCNativeHandle /*handle*/, uint32_t /*x*/,
-                                 uint32_t /*y*/, uint32_t /*width*/,
-                                 uint32_t /*height*/, uint32_t * /*stride*/,
-                                 void ** /*map_data*/, size_t /*plane*/) {
+void *Gralloc1BufferHandler::Map(HWCNativeHandle handle, uint32_t x,
+                                 uint32_t y, uint32_t width,
+                                 uint32_t height, uint32_t * /*stride*/,
+                                 void ** map_data, size_t /*plane*/) {
+  auto gr_handle = (struct cros_gralloc_handle *)handle->imported_handle_;
+  if (!gr_handle) {
+    ETRACE("could not find gralloc drm handle");
+    return NULL;
+  }
+
+  int acquireFence = -1;
+  gralloc1_rect_t rect{};
+  rect.left = x;
+  rect.top = y;
+  rect.width = width;
+  rect.height = height;
+
+  gralloc1_device_t *gralloc1_dvc =
+      reinterpret_cast<gralloc1_device_t *>(device_);
+  uint32_t status = lock_(gralloc1_dvc, handle->imported_handle_, GRALLOC_USAGE_SW_READ_RARELY,
+          GRALLOC_USAGE_SW_READ_RARELY, &rect, map_data, acquireFence);
+  ALOGI("%s - lock returned: %u", __FUNCTION__, status);
   return NULL;
 }
 
-void Gralloc1BufferHandler::UnMap(HWCNativeHandle /*handle*/,
+void Gralloc1BufferHandler::UnMap(HWCNativeHandle handle,
                                   void * /*map_data*/) {
+  auto gr_handle = (struct cros_gralloc_handle *)handle->imported_handle_;
+  if (!gr_handle) {
+    ETRACE("could not find gralloc drm handle");
+    return;
+  }
+
+  int releaseFence = 0;
+  gralloc1_device_t *gralloc1_dvc =
+      reinterpret_cast<gralloc1_device_t *>(device_);
+  unlock_(gralloc1_dvc, handle->imported_handle_, &releaseFence);
```

Comment:
Similarly we can return the status here.

Reply:
this API is inheriting from NativeBufferHandler. gbm_bo_umap is a void function too